javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/config/ConfigurationFileLoader.java

```diff
@@ -76,8 +79,36 @@ static Map<String, String> loadConfigFile() {
           configurationFilePath);
     }
 
+    StringSubstitutor strSubstitutor =
+        new StringSubstitutor(getStringLookup(configurationFilePath, configProperties));
+
     return properties.entrySet().stream()
-        .collect(Collectors.toMap(e -> e.getKey().toString(), e -> e.getValue().toString()));
+        .collect(
+            Collectors.toMap(
+                e -> e.getKey().toString(), e -> strSubstitutor.replace(e.getValue().toString())));
+  }
+
+  private static StringLookup getStringLookup(
+      String configurationFilePath, ConfigProperties configProperties) {
+    return placeholder -> {
+      String systemProperty = System.getProperty(placeholder);
+      if (systemProperty != null) {
+        return systemProperty;
+      }
+      String enVar = System.getenv(placeholder);
+      if (enVar != null) {
+        return enVar;
+      }
+      String configProperty = configProperties.getString(placeholder);
+      if (configProperty != null) {
+        return configProperty;
+      }
+      logger.log(
```

Comment:
Should this actually throw an exception and fail the agent load? 
I logged an error as this is what is done above if the properties file is unparseable, but I would generally favour a hard fail if the provided config is not being applied

Reply:
> Should this actually throw an exception and fail the agent load?

The general approach this project takes is "do no harm". We'd prefer to log a warning and continue operating just as if the file was not provided at all.